### PR TITLE
Distributor: move the ordering exemplars from ingester to distributor

### PR DESCRIPTION
**Pull request by ying-jeanne**
#### What this PR does
Currently, the exemplars are sorted in ingesters. However, since we're already iterating through exemplars in the distributor, it would be more efficient to perform this ordering earlier in the process. By doing so, we could achieve the ordering just once instead of three separate times.

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [x] Tests updated.
- [x] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [x] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Review by pstibrany — APPROVED** [review]
lgtm. Let's wait for @krajorama's review before merging, since he introduced the change.

**Review by 56quarters — APPROVED** [review]
LGTM, nice improvement. Can you add this PR number to the changelog entry that originally sorted exemplars or add a new entry if there's already been a release since then?

**Pull request merged**
